fcrepo-ldpath/src/main/java/org/fcrepo/camel/ldpath/ClientFactory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.camel.ldpath;
+
+import static java.util.Collections.emptyList;
+
+import java.util.List;
+
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.Credentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.marmotta.ldclient.api.endpoint.Endpoint;
+import org.apache.marmotta.ldclient.api.provider.DataProvider;
+import org.apache.marmotta.ldclient.endpoint.rdf.LinkedDataEndpoint;
+import org.apache.marmotta.ldclient.model.ClientConfiguration;
+import org.apache.marmotta.ldclient.provider.rdf.CacheProvider;
+import org.apache.marmotta.ldclient.provider.rdf.LinkedDataProvider;
+import org.apache.marmotta.ldclient.provider.rdf.RegexUriProvider;
+import org.apache.marmotta.ldclient.provider.rdf.SPARQLProvider;
+
+/**
+ * A convenience factory for creating a ClientConfiguration object
+ * @author acoburn
+ */
+public class ClientFactory {
+
+    /**
+     * Configure a linked data client suitable for use with a Fedora Repository.
+     * @param fedoraEndpoint a FedoraEndpoint configuration
+     * @return a configuration for use with an LDClient
+     */
+    public static ClientConfiguration createClient(final Endpoint fedoraEndpoint) {
+        return createClient(null, null, fedoraEndpoint, emptyList(), emptyList());
+    }
+
+    /**
+     * Configure a linked data client suitable for use with a Fedora Repository.
+     * @param fedoraEndpoint a FedoraEndpoint configuration
+     * @param endpoints additional endpoints to enable on the client
```

Comment:
And I think it's totally fine for those who want to use that style to use it. I have in hand the beginnings of a Fedora reimp (which of course we hope will be one of many). If I have to deal with the same code style that the `fcrepo4` uses, I will not be using `fcrepo4-exts` for it.